### PR TITLE
push to public docker repo

### PR DESCRIPTION
**Pull request by ninaburg**
To be accessible from a VM on the EWC, we need to push the image to the public docker registry.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -227,10 +235,14 @@ pipeline {
                         withCredentials([usernamePassword(credentialsId: 'openshift-nexus',
                             passwordVariable: 'NXPASS', usernameVariable: 'NXUSER')]) {
                             sh """
-                            echo $NXPASS | podman login ${Globals.IMAGE_REPO} -u $NXUSER --password-stdin
-                            podman push ${Globals.imageTag}
+                            echo $NXPASS | podman login ${Globals.IMAGE_REPO_INTERN} -u $NXUSER --password-stdin
+                            echo $NXPASS | podman login ${Globals.IMAGE_REPO_PUBLIC} -u $NXUSER --password-stdin
```
Comment (cfkanesan) [inline Jenkinsfile]: This could be moved into the conditional as well

File: Jenkinsfile [hunk 2]
```diff
@@ -266,13 +279,13 @@ pipeline {
...
-                    sh "echo $NXPASS | podman login ${Globals.IMAGE_REPO} -u $NXUSER --password-stdin"
-                    runDevScript("test/trivyscanner.py ${Globals.imageTag}")
+                    sh "echo $NXPASS | podman login ${Globals.IMAGE_REPO_INTERN} -u $NXUSER --password-stdin"
+                    runDevScript("test/trivyscanner.py ${Globals.imageTagIntern}")
                 }
             }
             post {
                 cleanup {
-                    sh "podman logout ${Globals.IMAGE_REPO} || true"
+                    sh "podman logout ${Globals.IMAGE_REPO_INTERN} || true"
```
Comment (cfkanesan) [inline Jenkinsfile]: I believe you also need to logout from the public repo

File: Jenkinsfile [hunk 3]
```diff
@@ -126,11 +129,13 @@ pipeline {
                         }
                         echo "Using version ${Globals.version}"
                         if (env.BRANCH_NAME == 'main') {
-                            Globals.imageTag = "${Globals.IMAGE_NAME}:latest"
+                            Globals.imageTagIntern = "${Globals.IMAGE_NAME_INTERN}:latest"
+                            Globals.imageTagPublic = "${Globals.IMAGE_NAME_PUBLIC}:latest"
```
Comment (cfkanesan) [inline Jenkinsfile:133]: I don't really have a full view on the deployment strategy but I have the impression that this might cause some issues with caching and the ability to control which image gets deployed. It's good from the point of view that amount storage required will not grow with time though. That said, this could be solved in a different PR.

Comment (ninaburg) [reply Jenkinsfile:133]: For the moment I'm mainly using the stage "pushing the artifacts" to run them on the VM and not the "deploy" stage, which fails.

Comment (cfkanesan) [reply Jenkinsfile:133]: That's fine. I was not referring to the deploy stage in the pipeline. At the moment, even if you deploy manually, you will need to tell the cache layer somehow that there is a new image tagged `latest` such that it will download the new image.

**Review by cfkanesan — APPROVED** [review]
